lib/Service/IpV6Strategy.php

```diff
@@ -50,8 +51,12 @@ public function findHistoricAndRecent(LoginAddressAggregatedMapper $loginAddress
 	}
 
 	protected function ipToVec(string $ip): array {
-		$addr = IPv6::factory($ip);
-		$hex = bin2hex($addr->getBinary());
+		$addr = inet_pton($ip);
+		if ($addr === false) {
+			throw new InvalidArgumentException('Invalid IPv6 address');
```

Comment:
@ChristophWurst Maybe you have a better idea on how to handle this a bit more gracefully?

Reply:
Doesn't look like `inet_pton` can fail on anything other than a truly invalid IP address value provided to it... Which seems highly unlikely given the source: either `REMOTE_ADDR` as provided directly by the web server (based on the functioning client IP connection) or from a proxy provided header (set also based on the actual connection there from the client). 

So in the unlikely event it does fail, something is **very very** wrong (so a hard exception seems desirable + acceptable IMO).

Reply:
I think this approach is fine. `\Darsyn\IP\Version\IPv6::factory` would through too :+1: